sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1659,6 +1659,12 @@ object SQLConf {
         "a SparkConf entry.")
       .booleanConf
       .createWithDefault(true)
+
+  val TIMESTAMP_EXTERNAL_TYPE = buildConf("spark.sql.catalyst.timestampType")
```

Comment:
We can support reading from both types at the same time right?
I don't know if it's worth changing what it is written to; not worth a flag IMHO.

Reply:
> We can support reading from both types at the same time right?

At Spark side, we can read both.

> I don't know if it's worth changing what it is written to; not worth a flag IMHO.

Timestamps can be loaded from a datasource, casted from other types and etc. If an user wants to imports (collect) non-legacy timestamps (I mean `java.time.Instant`), how she/he can do that without the flag?

Reply:
Import is fine; we could potentially read both types to TimestampType. Can we just be opinionated about the right way to write it back out, and keep current behavior? it may be 'legacy' but not sure it's worth the behavior change. You may have more context on why that's important though.

As with many things I just don't know how realistically people will understand the issue, find the flag, set it, and maintain it across deployments.

Reply:
> it may be 'legacy' but not sure it's worth the behavior change.

The SQL config `spark.sql.catalyst.timestampType` has default value `Timestamp` which preserves current behavior. When an user wants to import `java.time.Instant` from Spark, she/he can change the config to point out the Java timestamp class.